tiledb/api/c_api/profile/profile_api.cc

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file tiledb/api/c_api/profile/profile_api.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines C API functions for the profile section.
+ */
+
+#include "../error/error_api_internal.h"
+#include "../string/string_api_internal.h"
+#include "profile_api_internal.h"
+#include "tiledb/api/c_api_support/c_api_support.h"
+
+namespace tiledb::api {
+
+capi_return_t tiledb_profile_alloc(
+    const char* name, const char* homedir, tiledb_profile_t** profile) {
+  ensure_output_pointer_is_valid(profile);
+  if (!name) {
+    // Passing nullptr resolves to the default case.
+    name = tiledb::sm::RestProfile::DEFAULT_NAME.c_str();
+  } else if (name[0] == '\0') {
+    throw CAPIException("[tiledb_profile_alloc] Name cannot be empty.");
+  }
+
+  if (!homedir) {
+    // Passing nullptr resolves to the default case.
+    homedir = tiledb::common::filesystem::home_directory().c_str();
+  } else if (homedir[0] == '\0') {
+    throw CAPIException(
+        "[tiledb_profile_alloc] $HOME directory cannot be empty.");
```

Comment:
Writing `$HOME` in the error message is a little misleading, IMO this should refer to the named argument just like the above error does.
Consider changing both actually to something like `argument 'name' cannot be empty`